packages/vite/src/node/plugins/workerImportMetaUrl.ts

```diff
@@ -13,8 +13,63 @@ import { parseRequest } from '../utils'
 import { ENV_PUBLIC_PATH } from '../constants'
 import MagicString from 'magic-string'
 
+type WorkerType = 'classic' | 'module' | 'ignore'
+
 const WORKER_FILE_ID = 'worker_url_file'
 
+// find the workerOptions end with `)`
+function lexWorkerOptions(code: string, pos: number) {
+  let pattern = ''
+  const opStack = []
+
+  for (let i = pos; i < code.length; i++) {
+    const char = code.charAt(i)
+    if (char === '(') {
+      opStack.push(char)
+    } else if (char === ')') {
+      if (opStack.length) {
+        opStack.pop()
+      } else {
+        break
+      }
+    }
+    pattern += char
+  }
+
+  return pattern
+}
+
+function getWorkerType(code: string, i: number): WorkerType {
+  let workerOptsString = lexWorkerOptions(code, i)
+  const hasViteIgnore = /\/\*\s*@vite-ignore\s*\*\//.test(workerOptsString)
+  if (hasViteIgnore) {
+    return 'ignore'
+  }
+
+  workerOptsString = workerOptsString
+    .replace(multilineCommentsRE, '')
+    .replace(singlelineCommentsRE, '')
+
+  const match = /type\s*\:\s*(['"`]([^'|^"|^`]*)['"`]|.*)/.exec(
```

Comment:
For parsing `{ assert: { type: 'raw' } }` we are using JSON5, see https://github.com/vitejs/vite/blob/33f96718dc5d827612c300fb6d0258f1c040f5a1/packages/vite/src/node/importGlob.ts#L191. Given that we already have this dependency we may want to use it instead of the regex to parse the `{ type: 'module' }` object here. This should make the code more robust in the presence of comments (that I hope people will not be adding, but who knows). A parsing error in case of a dynamic type could be handled as an error that needs to be `@vite-ignored`

Reply:
I had try to use JSON5 🙈, but we can't distinguish the situation.

```js
{
  name: fn(),
  type: 'classic'
}
```

JSON5 will got a error, but it is not a dynamic type.

Reply:
Do we support this use case? Shouldn't we error or ask for `@vite-ignore`?

Reply:
I think we need to support, because `@vite-ignore` don't inject the env in dev mode (we don't know how import the `/@vite/env` file).

Reply:
> we don't know how import the `/@vite/env` file
I think of a way to solve this problem, we can copy the code of `/@vite/env` into worker file header. 😂